File: au-java/src/main/java/com/fran6k/study/springbootstudy/bean/Ass.java
package com.fran6k.study.springbootstudy.bean;

import com.alibaba.fastjson.JSONObject;
import com.fran6k.study.springbootstudy.bean.WxBean.Wxuser;
import lombok.Data;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "ass")
@Data
public class Ass {
    @Id
    @GeneratedValue(strategy= GenerationType.AUTO)
    //社团id
    private int assid;

    private String logo;

    //社团名称
    private String assname;
    //社团简介
    private String assDescription;
    //社团类型
    private String asstype;

    //社团经费
    private double money;
    //优秀社团
    private int isExecllent;
    //社长名称
    private String presidentname;

    //社长学号
    private String presidentId;

    //指导老师姓名
    private String teachername;

    //指导老师联系方式
    private String teacherphone;

    //指导老师所在单位
    private String teacherpost;

    //关联表
    @OneToMany(targetEntity = User.class)
    //从表中的外键+对应的主表中的主键
    @JoinColumn(name="assEntity",referencedColumnName =  "assid")
    private List<User> user;

    public List<User> getUserList(){
        return user;
    }

    public JSONObject getAss(){
        JSONObject jsonObject = new JSONObject();
        jsonObject.put("assId",assid);
        jsonObject.put("logo",logo);
        jsonObject.put("assName",assname);
        jsonObject.put("assType",asstype);
        jsonObject.put("assDescription",assDescription);
        jsonObject.put("money",money);
        jsonObject.put("isExecllent",isExecllent);
        jsonObject.put("presidentName",presidentname);
        jsonObject.put("presidentId",presidentId);
        jsonObject.put("teacherName",teachername);
        jsonObject.put("teacherPhone",teacherphone);
        jsonObject.put("teacherPost",teacherpost);
        return jsonObject;
    }
}

File: au-association-go/model/system/Ass.go
package system

type Ass struct {
	// 社团id
	Assid int    `json:"assId" form:"assId" gorm:"primary_key"`
	Logo  string `json:"logo" form:"-"`
	// 社团名称
	Assname string `json:"assName" form:"assName"`
	// 社团简介
	AssDescription string `json:"assDescription" form:"assDescription"`
	// 社团类型
	Asstype string `json:"assType" form:"assType"`
	// 社团经费
	Money float64 `json:"money" form:"money"`
	// 优秀社团
	IsExecllent int `json:"isExecllent" form:"isExecllent"`
	//社长名称
	Presidentname string `json:"presidentName" form:"presidentName"`
	// 社长学号
	PresidentId string `json:"presidentId" form:"presidentId"`
	// 指导老师姓名
	Teachername string `json:"teacherName" form:"teacherName"`
	// 知道老师联系方式
	Teacherphone string `json:"teacherPhone" form:"teacherPhone"`
	// 指导老师所在单位
	Teacherpost string `json:"teacherPost" form:"teacherPost"`

	//AssUsers []User `gorm:"foreignKey:AssEntity;references:Assid"` //后台系统用户

	//AssWxUsers []WxBean.WxUser `gorm:"foreignKey:AssEntityFirst;foreignKey:AssEntitySecond;AssociactionPrimariKey:Assid"`
}

func (Ass) TableName() string {
	return "ass"
}
